tasks/base.py

```diff
@@ -42,49 +42,58 @@ def metrics_prefix(self):
 
     def on_timeout(self, soft: bool, timeout: int):
         res = super().on_timeout(soft, timeout)
+        task_group = (
+            self.name.split(".")[-2] if self.name is not None else "unknown_group"
```

Comment:
[nit] this could be in a function since you are doing it somewhere else as well

Reply:
also are we positive all task names definitely fit this pattern?

Reply:
They are supposed to fit this pattern. It's the pattern we use for the [task routes in celery config](https://github.com/codecov/shared/blob/main/shared/celery_config.py)

I can add a try/except just in case